File: app/src/main/java/com/robotnec/budget/core/di/module/NavigationModule.java
package com.robotnec.budget.core.di.module;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import com.robotnec.budget.core.navigator.Navigator;

/**
 * @author zak <zak@swingpulse.com>
 */
@Module
public class NavigationModule {

    private Navigator navigator;

    public NavigationModule(Navigator navigator) {
        this.navigator = navigator;
    }

    @Singleton
    @Provides
    public Navigator provideInternalNavigator() {
        return navigator;
    }
}

File: app/src/main/java/com/robotnec/budget/core/di/module/NavigationModule.kt
package com.robotnec.budget.core.di.module

import com.robotnec.budget.app.navigator.AndroidNavigator
import com.robotnec.budget.core.navigator.Navigator
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * @author zak zak@swingpulse.com>
 */
@Module
class NavigationModule {

    @Singleton
    @Provides
    fun provideInternalNavigator(): Navigator {
        return AndroidNavigator()
    }
}
